Formula/d/distcc.rb

```diff
@@ -23,42 +25,52 @@ class Distcc < Formula
     sha256 x86_64_linux:   "afcb3e0911ffdb494af0fb80013de9c990690812e86f77001dde497ee0966b79"
   end
 
-  depends_on "autoconf" => :build
-  depends_on "automake" => :build
-  depends_on "python-setuptools"
   depends_on "python@3.12"
 
   resource "libiberty" do
     url "https://ftp.debian.org/debian/pool/main/libi/libiberty/libiberty_20210106.orig.tar.xz"
     sha256 "9df153d69914c0f5a9145e0abbb248e72feebab6777c712a30f1c3b8c19047d4"
   end
 
+  resource "setuptools" do
+    url "https://files.pythonhosted.org/packages/d6/4f/b10f707e14ef7de524fe1f8988a294fb262a29c9b5b12275c7e188864aed/setuptools-69.5.1.tar.gz"
+    sha256 "6c1fccdac05a97e598fb0ae3bbed5904ccb317337a51139dcd51453611bbb987"
+  end
+
   # Python 3.10+ compatibility
   patch do
     url "https://github.com/distcc/distcc/commit/83e030a852daf1d4d8c906e46f86375d421b781e.patch?full_index=1"
     sha256 "d65097b7c13191e18699d3a9c7c9df5566bba100f8da84088aa4e49acf46b6a7"
   end
 
+  # Switch from distutils to setuptools
+  patch do
+    url "https://github.com/distcc/distcc/commit/76873f8858bf5f32bda170fcdc1dfebb69de0e4b.patch?full_index=1"
+    sha256 "611910551841854755b06d2cac1dc204f7aaf8c495a5efda83ae4a1ef477d588"
+  end
+
   def install
     ENV["PYTHON"] = python3 = which("python3.12")
     site_packages = prefix/Language::Python.site_packages(python3)
 
+    build_venv = virtualenv_create(buildpath/"venv", python3)
+    build_venv.pip_install resource("setuptools")
+    ENV.prepend_create_path "PYTHONPATH", build_venv.site_packages
+
     # While libiberty recommends that packages vendor libiberty into their own source,
     # distcc wants to have a package manager-installed version.
     # Rather than make a package for a floating package like this, let's just
     # make it a resource.
-    buildpath.install resource("libiberty")
-    cd "libiberty" do
-      system "./configure"
-      system "make"
+    resource("libiberty").stage do
+      system "./libiberty/configure", "--prefix=#{buildpath}", "--enable-install-libiberty"
+      system "make", "install"
     end
-    ENV.append "LDFLAGS", "-L#{buildpath}/libiberty"
-    ENV.append_to_cflags "-I#{buildpath}/include"
+    ENV.append "CPPFLAGS", "-I#{buildpath}/include"
+    ENV.append "LDFLAGS", "-L#{buildpath}/lib"
 
     # Make sure python stuff is put into the Cellar.
     # --root triggers a bug and installs into HOMEBREW_PREFIX/lib/python2.7/site-packages instead of the Cellar.
```

Comment:
Maybe the comment is worth updating too, Python 2.7 is now long gone :)

Reply:
agreed!